src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+A simple string interpolation system. Accepts a template string with embedded tokens of the form `%value`. Returns a new string with the tokens replaced by the specified values.
+Keys and values can be of any data type that can be coerced into a string. Tokens are not delimited in the template with any syntax - the implementation will replace the longest token which
+can be sourced from `terms`.
+
+* `template {String}` A string with embedded tokens of the form `%value`
+* `terms {Object: String → Stringable}` A hash which maps tokens to their required interpolated values   
+
+
+## Handling models and paths
+
+### fluid.get(model, path)
+
+Accesses an element nested within a [model object](FrameworkConcepts.md#model-object] at the supplied (EL) path. `fluid.get` can
+operate various special behaviour in encountering particular kinds of material - for example, any object implementing a function named `resolvePathSegment` will
+hand resolution of the next path segment to that function if there is no corresponding concrete member. If `fluid.get` navigates off the end of the available
+model material, this will not trigger an error - all successive indirections will simply return `undefined`. `fluid.get` accepts various esoteric piece of configuration
+allowing it to operate indirection rules encoded in arbitrary JSON material - these are imperfectly supported, contact us if you are interested.
+
+* `model {Any}` The model object to be indirected into. This may be `undefined`, etc. without triggering an error.
+* `path {String|Array of String}` Either a period-separated [EL path](FrameworkConcepts.md#el-paths) representing a chain of properties to be navigated, or else an array of
+path segments representing this
+* Returns: `{Any}` The result of the indirection of the path into the supplied model
+
+### fluid.getImmediate(model, path) 
+
+An extremely simplistic, high-performance variant of `fluid.get` that supports none of the same bells and whistles - it is only capable of navigating over concrete
+object properties that are actually present in the model, and the path must have already been parsed out into an array of segments. 
+It still, however, operates the same semantic as `fluid.get` for navigating over undefined values.
+
+* `model {Any}` The model object to be indirected into. This may be `undefined`, etc. without triggering an error.
+* `path {Array of String}` An array of path segments representing representing a chain of properties to be navigated
+* Returns: `{Any}` The result of the indirection of the path into the supplied model
+
+### fluid.set(model, path, newValue)
+
+Writes a new value into a model at a specified path. This is performed by directly modifying the penultimately referenced model object (the one referenced by the first `n-1` 
+path segments in `path`) by writing the value `newValue` to its property named after the final path segment. Other approaches to this task are possible 
```

Comment:
Yes, perish the thought that somebody might try to assemble a mental model of what they are doing as a result of reading this documentation : P